packages/flutter/lib/src/material/tabs.dart

```diff
@@ -557,8 +558,18 @@ class _TabBarState extends State<TabBar> {
     });
   }
 
+  // Called each time layout completes.
   void _saveTabOffsets(List<double> tabOffsets) {
     _indicatorPainter?.tabOffsets = tabOffsets;
+
+    if (!_completedInitialScroll) {
+      _completedInitialScroll = true;
+      if (widget.isScrollable && _currentIndex != 0) {
+        scheduleMicrotask(() {
```

Comment:
PageView creates subclasses of ScrollController/ScrollPosition and appears to use ScrollPosition.correctPixels() to fixup the  scroll offset once the viewport dimensions are determined.  At least that's what appears to be going on. 

I'll try the correctPixels() approach.